src/storage/page_buffer.c

```diff
@@ -2389,6 +2368,29 @@ pgbuf_check_page_ahead_of_replication (THREAD_ENTRY * thread_p, PAGE_PTR page)
 }
 #endif
 
+int
+pgbuf_check_for_deallocated_page_or_desyncronization (THREAD_ENTRY * thread_p, PGBUF_LATCH_MODE latch_mode,
+						      const VPID & vpid)
+{
+  int ret = ER_PB_BAD_PAGEID;
```

Comment:
If bad pageid error is assumed, better to safe-guard that it is true.
```suggestion
  assert (er_errid () == ER_PB_BAD_PAGEID);
  int ret = ER_PB_BAD_PAGEID;
```